src/bun.js/ipc.zig

```diff
@@ -156,17 +156,162 @@ pub const IPCData = struct {
     }
 };
 
-/// This type is shared between VirtualMachine and Subprocess for their respective IPC handlers
-///
-/// `Context` must be a struct that implements this interface:
-/// struct {
-///     globalThis: ?*JSGlobalObject,
-///     ipc: IPCData,
-///
-///     fn handleIPCMessage(*Context, DecodedIPCMessage) void
-///     fn handleIPCClose(*Context, Socket) void
-/// }
-pub fn NewIPCHandler(comptime Context: type) type {
+const NamedPipeIPCData = struct {
+    const uv = bun.windows.libuv;
+
+    pipe: uv.uv_pipe_t,
+    incoming: bun.ByteList = .{}, // Maybe we should use IPCBuffer here as well
+    outgoing: IPCBuffer = .{},
+    connected: bool = false,
+    has_written_version: if (Environment.allow_assert) u1 else u0 = 0,
+    connect_req: uv.uv_connect_t = std.mem.zeroes(uv.uv_connect_t),
+    server: uv.uv_pipe_t = std.mem.zeroes(uv.uv_pipe_t),
+    current_payload_len: usize = 0,
+
+    pub fn processSend(this: *NamedPipeIPCData) void {
+        const bytes = this.outgoing.list.slice()[this.outgoing.cursor..];
+        log("processSend {d}", .{bytes.len});
+        if (bytes.len == 0) return;
+
+        const req = bun.new(uv.uv_write_t, std.mem.zeroes(uv.uv_write_t));
+        req.data = @ptrCast(this);
+        req.write_buffer = uv.uv_buf_t.init(bytes);
+        log("processSend write_buffer {d}", .{req.write_buffer.len});
+        this.current_payload_len = bytes.len;
+        const write_err = uv.uv_write(req, @ptrCast(&this.pipe), @ptrCast(&req.write_buffer), 1, NamedPipeIPCData.uvWriteCallback).int();
+        if (write_err < 0) {
+            Output.printErrorln("Failed write IPC version", .{});
+            return;
+        }
+    }
+
+    fn uvWriteCallback(req: *uv.uv_write_t, status: uv.ReturnCode) callconv(.C) void {
```

Comment:
If we do use this approach - `uv.uv_write`, we should have a wrapper so the arguments are `this: *NamedIPCPipeData` instead of having to deal with the internals of libuv everywhere

```suggestion
    fn onWriteComplete(this: *NamedIPCPipeData, result: union(enum) { written: u32, err: bun.sys.Error}) callconv(.C) void {
```

Reply:
We dont have a reliable internal way to see how many bytes was written only using the libuv struct (to avoid allocations) on windows, so I did `fn onWriteComplete(this: *NamedPipeIPCData, status: uv.ReturnCode) void` but is already way better